src/app/clusters/mode-base-server/mode-base-server.h

```diff
@@ -0,0 +1,198 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "mode-base-cluster-objects.h"
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/util/af.h>
+#include <map>
+
+using chip::Protocols::InteractionModel::Status;
+using ModeOptionStructType = chip::app::Clusters::detail::Structs::ModeOptionStruct::Type;
+using ModeTagStructType    = chip::app::Clusters::detail::Structs::ModeTagStruct::Type;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ModeBase {
+
+class Instance : public CommandHandlerInterface, public AttributeAccessInterface
+{
+public:
+    // This map holds pointers to all initialised ModeBase instances. It provides a way to access all ModeBase clusters.
+    static std::map<uint32_t, Instance *> ModeBaseAliasesInstanceMap;
+
+    /**
+     * This is a helper function to build a mode option structure. It takes the label/name of the mode,
+     * the value of the mode and a list of mode tags that apply to this mode. NOTE, the caller must
+     * ensure that the lifetime of the label and modeTags is as long as the returned structure.
+     */
+    static ModeOptionStructType BuildModeOptionStruct(const char * label, uint8_t mode,
+                                                      const List<const ModeTagStructType> modeTags)
+    {
+        detail::Structs::ModeOptionStruct::Type option;
+        option.label    = CharSpan::fromCharString(label);
+        option.mode     = mode;
+        option.modeTags = modeTags;
+        return option;
+    }
+
+    CHIP_ERROR Init();
```

Comment:
How can this fail? Please document. If it cannot fail, please make it return void.

Reply:
Done [here](https://github.com/project-chip/connectedhomeip/pull/27504/commits/cbc7401ca9e94c6e835bed84cf97151ecf010feb).